### ISSUE TITLE
[FG:InPlacePodVerticalScaling] ResourceQuota unresponsive to scale-down

### ISSUE THREAD

**Issue by tallclair** [kubernetes/kubernetes#127132]
/kind bug

**TL;DR:**

If a pod's requests are scaled down with InPlacePodVerticalScaling, it can take a long time (up to 5 minutes by default) for resource quota to free the resources. If the pod is scaled back up in that window, the difference will be double-counted.

**Explanation:**

If a pod is resized, the resource quota admission controller relies on the [resource helper `resource.PodRequests`](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/resource/helpers.go#L50) to determine the total request size. That function uses the maximum of the resource request (`.spec.containers[*].resources.requests`) and the allocated resources (`.status.containerStatuses[*].allocatedResources`).

In the event of scaling down, the allocated resource value will be larger, and therefore the value that is used to set the resource quota. This is working as intended, but the ResourceQuota controller should pick up the scale down once the Kubelet updates the allocated resources.

The resource quota's default filter function needs to be updated to consider updates to allocated resources:
https://github.com/kubernetes/kubernetes/blob/48d6d55a47ca8202692098639f7b3c76f107f46a/pkg/quota/v1/install/update_filter.go#L26-L33

Currently, the status update is ignored, so resource quota never sees the actual scale down, and the quota is only updated by the periodic resync.

/sig node
/milestone v1.32
/priority important-longterm

**Comment by bouaouda-achraf** [comment]
If triage accepted I can work on it ?✋
Thanks

**Comment by tallclair** [comment]
/triage accepted
/assign @bouaouda-achraf 
:slightly_smiling_face:

**Comment by CodeDiego15** [comment]
Hi, Can I work on it?

**Comment by wendy-ha18** [comment]
Hello @haircommander @bouaouda-achraf @tallclair 

Friendly reminder that code freeze is starting 02:00 UTC Friday November 8th 2024 (about 3 weeks from now), and while there is still time, we want to ensure that each PR has a chance to be merged on time.

Given this timeline and capacity, will this continue to be aimed for the 1.32 release?
Thanks! 😄

**Comment by wendy-ha18** [comment]
Hey @haircommander @bouaouda-achraf @tallclair
As this PR is tagged for the v1.32 release, just a last reminder from my side that the code freeze is starting 02:00 UTC Friday November 8th 2024 (less than 1 week from now). 

Please make sure the PR has both lgtm and approved labels before the code freeze. Thanks!
